src/backend/utils/adt/dbsize.c

```diff
@@ -180,9 +180,7 @@ calculate_database_size(Oid dbOid)
 	Assert(Gp_role != GP_ROLE_EXECUTE);
 
 	if (dbOid == HcatalogDbOid)
-		ereport(ERROR,
-			(ERRCODE_UNDEFINED_DATABASE,
-			errmsg("database hcatalog (OID 6120) is reserved")));
+		return -1;
```

Comment:
This is to solve incompatibility with psql clients (not hawq client). In hawq client, we hide hcatalog from being displayed in \l or \l+. But with other psql clients, if we error out in dbsize, \l and \l+ will also error out and show nothing.

A compromise solution, which we implemented here is to return hcatalog dbsize as -1. Based on my investigation and interview with some users they think -1 is better than 0, as it's not a valid size info, so users who may rely on this value can adjust their logic to exclude hcatalog as necessary.